infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/TrackLayout.kt

```diff
@@ -189,6 +191,34 @@ data class TrackLayoutKmPost(
     val exists = !state.isRemoved()
 }
 
+data class TrackLayoutKmPostLengthDetails(
+    val trackNumberId: IntId<TrackLayoutTrackNumber>,
+    val kmNumber: KmNumber,
+    val startM: BigDecimal,
+    val endM: BigDecimal,
+    val locationSource: GeometrySource,
+    val location: IPoint? = null,
+) {
+    val length = endM - startM
+
+    init {
+        require(startM.scale() == 3) {
```

Comment:
Ehkä turha require, siellä missä luodaan näitä on aina roundTo3Decimals

Reply:
Tosiaan vähän mielenkiintoinen. Toinen tapa hoitaa tämä olisi, että hoidettaisiin se rounding itsessäänkin jo täällä. Tällöin tulisi mielestäni vähän selkeämpi vastuujakokin, eikä tämän tarvisi varmistaa että roundasihan aiempi koodi luvun varmasti valmiiksi